plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/AccountsPanel.java

```diff
@@ -82,13 +86,19 @@ protected Control createDialogArea(Composite parent) {
     GridLayoutFactory.swtDefaults().generateLayout(container);
 
     createAccountsPane(container);
-    createButtons(container);
     return container;
   }
 
   @VisibleForTesting
   void createAccountsPane(Composite accountArea) {
-    for (Account account : loginService.getAccounts()) {
+    List<Account> accounts;
+    try {
+      accounts = Collections.singletonList(apiFactory.getAccount());
+    } catch (IOException ex) {
+      logger.log(Level.SEVERE, "Could not obtain ADC account", ex);
+      accounts = Collections.emptyList();
```

Comment:
Syntax sugar (meaning: your choice, but consider for readability) -- a private `getAccounts()` method would prevent needing to have this relatively clunky "declare then initialize then use" order of events.

```java
private List<Account> getAccounts() {
  try {
    return Collections.singletonList(apiFactory.getAccount());
  } catch (IOException ex) {
    ...
    return Collections.emptyList();
  }
}
void createAccountsPane(...) {
  for (Account account : getAccounts()) {
    ...
  }
}
```